.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  release:
+    types: [published]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+  REGISTRY: ghcr.io
+  RUST_VERSION: 1.61.0
+
+jobs:
+  build-sway-lib:
+      runs-on: ubuntu-latest
+      
+      steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
+
+        - name: Install Rust toolchain
+          uses: actions-rs/toolchain@v1
+          with:
+            profile: minimal
+            toolchain: ${{ env.RUST_VERSION }}
+            override: true
+
+        - name: Init cache
+          uses: Swatinem/rust-cache@v1
+
+        - name: Install a modern linker (mold)
+          uses: rui314/setup-mold@v1
+
+        - name: Force Rust to use mold globally for compilation
+          run: |
+            touch ~/.cargo/config.toml
+            echo "[target.x86_64-unknown-linux-gnu]" > ~/.cargo/config.toml
+            echo 'linker = "clang"' >> ~/.cargo/config.toml
+            echo 'rustflags = ["-C", "link-arg=-fuse-ld=/usr/local/bin/mold"]' >> ~/.cargo/config.toml
+
+        - name: Install rustfmt
+          run: rustup component add rustfmt
+
+        - name: Install Fuel toolchain
+          uses: FuelLabs/action-fuel-toolchain@v0.1.0
+
+        - name: Check Sway formatting
+          run: forc fmt --path sway-libs --check
+
+        - name: Build sway-libs
+          run: forc build --path sway-libs

```

Comment:
```suggestion
        - name: Check Sway formatting
          run: forc fmt --path sway_libs --check

        - name: Build sway-libs
          run: forc build --path sway_libs

```

Reply:
I never fail to shoot myself in the foot